### PR TITLE
[Doc] Add SUN RGB-D doc

### PR DESCRIPTION
**Pull request by THU17cyz**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Add SUN RGB-D doc.

## Modification
- Add SUN RGB-D doc (mainly based on ScanNet doc).
- Modify ScanNet doc for consistency.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by THU17cyz** [comment]
TBD: Multi-modality related contents.

File: docs/datasets/sunrgbd_det.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+For the overall process, please refer to the [README](https://github.com/open-mmlab/mmdetection3d/blob/master/data/sunrgbd/README.md/) page for SUN RGB-D.
+
+### Download SUN RGB-D data and toolbox
+
+Download SUNRGBD data [HERE](http://rgbd.cs.princeton.edu/data/). Then, move SUNRGBD.zip, SUNRGBDMeta2DBB_v2.mat, SUNRGBDMeta3DBB_v2.mat and SUNRGBDtoolbox.zip to the OFFICIAL_SUNRGBD folder, unzip the zip files. Currently, we use v1 data for training and testing, so the data files with suffix `_v2` is unnecessary.
+
+The directory structure before data preparation should be as below：
+
+```
+```
```
Comment (wHao-Wu) [inline docs/datasets/sunrgbd_det.md:14]: duplicated ' \`\`\` '

File: docs/datasets/sunrgbd_det.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+│   ├── SUNRGBDtoolbox
+├── sunrgbd_trainval
+│   ├── calib
+│   ├── depth
+│   ├── image
+│   ├── label
+│   ├── label_v1
+│   ├── seg_label
+```
+```
```
Comment (wHao-Wu) [inline docs/datasets/sunrgbd_det.md:34]: duplicated ' ``` '

File: docs/datasets/sunrgbd_det.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+Under each following directory there are overall 5285 train files and 5050 val files:
+
+- 'calib': Camera calibration information in `.txt`
```
Comment (wHao-Wu) [inline docs/datasets/sunrgbd_det.md:38]: Use \` ... \` instead of ' ... '

File: docs/datasets/sunrgbd_det.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+- `depth`: Point cloud saved in `.mat` (xyz+rgb)
+- `image`: Image data in `.jpg`
+- `label`: Detection nnotation data in `.txt` (version 2)
```
Comment (wHao-Wu) [inline docs/datasets/sunrgbd_det.md:41]: `nnotation` -> `annotation`

File: docs/datasets/sunrgbd_det.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+                [
+                    obj.box3d for obj in obj_list
+                    if obj.classname in self.cat2label.keys()
+                ],
+                axis=0)  # (K,8)
+        info['annos'] = annotations
+    return info
+```
+
+The directory structure after process should be as below
```
Comment (wHao-Wu) [inline docs/datasets/sunrgbd_det.md:169]: Changing `after process` to `after processing` will be better?

File: docs/datasets/sunrgbd_det.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+```
+
+- Data augmentation:
+    - `RandomFlip3D`: randomly flip input point cloud horizontally or vertically.
+    - `GlobalRotScaleTrans`: rotate input point cloud, usually in the range of [-30, 30] (degrees) for SUN RGB-D.
+    - `IndoorPointSample`: downsample input point cloud.
+
+## Metrics
+
+Same as ScanNet, typically mean Average Precision (mAP) is used for evaluation on SUN RGB-D, e.g. `mAP@0.25` and `mAP@0.5`. In detail, a generic functions to compute precision and recall for 3D object detection for multiple classes is called, please refer to [indoor_eval](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/core/evaluation/indoor_eval.py).
```
Comment (wHao-Wu) [inline docs/datasets/sunrgbd_det.md:255]: `a generic functions` -> `a generic function`

File: docs/datasets/scannet_det.md [hunk 7]
```diff
@@ -271,14 +271,16 @@ train_pipeline = [
...
+    - `GlobalRotScaleTrans`: rotate input point cloud, usually in the range of [-5, 5] (degrees) for ScanNet.
 
 ## Metrics
 
-Typically mean average precision (mAP) is used for evaluation on ScanNet, e.g. `mAP@0.25` and `mAP@0.5`. In detail, a generic functions to compute precision and recall for 3d object detection for multiple classes is called, please refer to [indoor_eval](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/core/evaluation/indoor_eval.py).
-As introduced in section `Export ScanNet data`, all ground truth 3d bounding box are axis-aligned, i.e. the yaw is zero. So the yaw target of network predicted 3d bounding box is also zero and axis-aligned 3d non-maximum suppression (NMS) is adopted during post-processing without reagrd to rotation.
+Typically mean Average Precision (mAP) is used for evaluation on ScanNet, e.g. `mAP@0.25` and `mAP@0.5`. In detail, a generic functions to compute precision and recall for 3D object detection for multiple classes is called, please refer to [indoor_eval](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3D/core/evaluation/indoor_eval.py).
```
Comment (wHao-Wu) [inline docs/datasets/scannet_det.md:284]: `a generic functions` -> `a generic function`

File: tools/data_converter/sunrgbd_data_utils.py [hunk 8]
```diff
@@ -191,9 +191,9 @@ def process_single_scene(sample_idx):
                     ],
                                                              axis=0)
                     annotations['dimensions'] = 2 * np.array([
-                        [obj.l, obj.h, obj.w] for obj in obj_list
+                        [obj.l, obj.w, obj.h] for obj in obj_list
```
Comment (Wuziyi616) [inline tools/data_converter/sunrgbd_data_utils.py:194]: If I understand correctly, this change will cause BC-breaking in the processed SUN RGB-D data? Why this PR which targets on docs will need to change this?

File: docs/datasets/sunrgbd_det.md [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+Under each following directory there are overall 5285 train files and 5050 val files:
+
+- `calib`: Camera calibration information in `.txt`
+- `depth`: Point cloud saved in `.mat` (xyz+rgb)
+- `image`: Image data in `.jpg`
+- `label`: Detection annotation data in `.txt` (version 2)
+- `label_v1`: Detection annotation data in `.txt` (version 1)
+- `seg_label`: Segmentation annotation data in `.txt`
+
+Extract SUN RGB-D annotation data by running:
```
Comment (Wuziyi616) [inline docs/datasets/sunrgbd_det.md:43]: Can we add a comment like `(this requires MATLAB installed on your machine)`

**Review by Wuziyi616 — COMMENTED** [review]
We also need to add this doc to index.rst. And please add the corresponding Chinese doc (An empty file containing a head title is OK)

File: tools/data_converter/sunrgbd_data_utils.py (hunk 8 above)
Comment (THU17cyz) [reply tools/data_converter/sunrgbd_data_utils.py:194]: No, this 'dimensions' is unused. I changed this because this is very confusing.

Comment (THU17cyz) [reply tools/data_converter/sunrgbd_data_utils.py:194]: When I added comments in the code fraction in the doc, I wrote ... boxes (in depth coordinate system)... However this 'dimensions' is in camera coordinate system

Comment (Wuziyi616) [reply tools/data_converter/sunrgbd_data_utils.py:194]: ICIC

**Review by Wuziyi616 — APPROVED** [review]
Almost ready for merge

**Comment by THU17cyz** [comment]
Made a mistake in the dataset preparation part. @Wuziyi616 Please go through that part again. It's been too long that I've processed the raw data orz

File: docs_zh-CN/datasets/sunrgbd_det.md [hunk 10]
```diff
@@ -0,0 +1,345 @@
+# 3D目标检测SUN RGB-D数据集
```
Comment (Wuziyi616) [inline docs_zh-CN/datasets/sunrgbd_det.md:1]: 中英文空格

File: docs_zh-CN/datasets/sunrgbd_det.md [hunk 11]
```diff
@@ -0,0 +1,345 @@
...
+mat_filename = strcat(num2str(imageId,'%06d'), '.mat');
+txt_filename = strcat(num2str(imageId,'%06d'), '.txt');
+% 保存点云数据
+parsave(strcat(depth_folder, mat_filename), points3d_rgb);
+```
+
+用于提取并组织检测任务标注的 `extract_rgbd_data_v1.m` 的主要部分如下：
+
+```matlab
+% 输出 2D 和 3D 包围盒
```
Comment (Wuziyi616) [inline docs_zh-CN/datasets/sunrgbd_det.md:69]: `包围盒` --> `包围框`？别的文档里似乎都是说的 `框`

File: docs_zh-CN/datasets/sunrgbd_det.md [hunk 12]
```diff
@@ -0,0 +1,345 @@
...
+    dict(type='Collect3D', keys=['points', 'gt_bboxes_3d', 'gt_labels_3d'])
+]
+```
+
+点云上的数据增强
+- `RandomFlip3D`：随机左右或前后翻转输入点云。
+- `GlobalRotScaleTrans`：旋转输入点云，对于 SUN RGB-D 角度通常落入 [-30, 30] （度）的范围；并放缩输入点云，对于 SUN RGB-D 比例通常落入 [0.85, 1.15] 的范围；最后平移输入点云，对于 SUN RGB-D 通常位移量为 0。
+- `IndoorPointSample`：降采样输入点云。
+
+SUN RGB-D 上多模态（点云和图像） 3D 物体检测的经典流程如下：
```
Comment (Wuziyi616) [inline docs_zh-CN/datasets/sunrgbd_det.md:295]: `）` 和 `3D` 之间不需要空格

File: docs_zh-CN/datasets/sunrgbd_det.md (hunk 11 above)
Comment (Wuziyi616) [reply docs_zh-CN/datasets/sunrgbd_det.md:69]: 下面的也请改一下吧

**Review by Wuziyi616 — COMMENTED** [review]
Some minor typos in Chinese doc

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**